server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -570,29 +627,44 @@ public int getTotal() {
          * @return how many total clusters the search was executed successfully on
          */
         public int getSuccessful() {
-            return successful;
+            if (clusterInfo.isEmpty()) {
```

Comment:
`clusterInfo` is always empty for local-only searches and CCS MRT=false, so it can be used as the proxy for when Clusters is immutable vs. updatable.

Reply:
I have an idea for refactoring. Will be in the next push.